tests/test_warc_metadata_sidecar.py

```diff
@@ -0,0 +1,144 @@
+                                 b'</head>\n'
+                                 b'<body>some text</body>\n'
+                                 b'</html>\n'
+                                 b'</html>\n')}
+
+CLD2 = (True,
+        7896,
+        (('Unknown', 'un', 99, 1070.0), ('Unknown', 'un', 0, 0.0), ('Unknown', 'un', 0, 0.0)))
+
+WARCINFO_DICT = {'software': 'warc-metadata-sidecar/1.0',
+                 'hostname': HOSTNAME,
+                 'ip': socket.gethostbyname(HOSTNAME),
+                 'conformsTo': 'http://bibnum.bnf.fr/WARC/WARC_ISO_28500_version1_latestdraft.pdf',
+                 'description': 'WARC metdata sidecar for sample.warc',
+                 'publisher': 'University of North Texas - Digital Projects Unit'}
+
+
+def test_find_mime_and_puid():
+    fido = sidecar.ExtendFido()
+    sidecar.find_mime_and_puid(fido, RECORD1['payload'], RECORD1['url'])
+    assert fido.puid == 'fmt/471'
+    assert fido.mime == 'text/html'
+
+
+def test_find_character_set():
+    RECORD1['payload'].seek(0)
+    decoded_payload = RECORD1['payload'].read()
+    result_dict = sidecar.find_character_set(decoded_payload)
+    assert result_dict == {'encoding': 'ascii', 'confidence': 1.0}
+
+
+def test_find_language():
+    RECORD1['payload'].seek(0)
+    decoded_payload = RECORD1['payload'].read()
+    language = sidecar.find_language(decoded_payload)
+    assert language == {'reliable': True,
+                        'text-bytes': 11,
+                        'languages': [{'name': 'ENGLISH',
+                                       'code': 'en',
+                                       'text-covered': 90,
+                                       'score': 2048.0}]}
+
+
+def test_unknown_language():
+    with patch.object(cld2, 'detect', return_value=CLD2):
+        language = sidecar.find_language(b'some_bytes')
+    assert language is None
+
+
+def test_create_warcinfo_payload():
+    publisher = 'University of North Texas - Digital Projects Unit'
+    warcinfo = sidecar.create_warcinfo_payload('sample.warc', None, publisher)
+    assert warcinfo == WARCINFO_DICT
+
+
+class Test_Warc_Metadata_Sidecar:
+
+    @patch('warc_metadata_sidecar.create_warcinfo_payload')
+    @patch('warc_metadata_sidecar.WARCWriter')
+    def test_metadata_sidecar(self, mock_warcwriter, m_warcinfo, caplog, tmpdir):
+        caplog.set_level(INFO)
+        writer = mock_warcwriter.return_value
+        sidecar.metadata_sidecar(str(tmpdir), TEXT_TEST_FILE)
+        assert 'Logging WARC metadata record information for %s', TEXT_TEST_FILE in caplog.text
+        assert 'Found 1 record(s)' in caplog.text
+        assert tmpdir / 'text.warc.meta.gz' in tmpdir.listdir()
+        assert writer.write_record.call_count == 2
+        m_warcinfo.assert_called_with('text.warc', None, None)
+        calls = [call(writer.create_warcinfo_record.return_value),
+                 call(writer.create_warc_record.return_value)]
+        writer.write_record.assert_has_calls(calls)
```

Comment:
Ideally we could check the information being written to the sidecar record for this test too...I know we are mocking the WARCWriter currently, so can't check a written WARC...is there a way to verify it with a check to the call of  writer.create_warc_record or is that too messy with payload being an io.BytesIO? Something to consider would be to break the `string_payload` building out to a function, so it would be easier to test the `string_payload` generating outside of `test_metadata_sidecar` and also shorten the code directly in `metadata_sidecar`. We can discuss what you think would be best to do regarding this comment.

Reply:
I like the idea of creating `string_payload` into a separate function. I tired asserting the return of writer.create_warc_record and I couldn't get it to match because of the `io.BytesIO` object.